### PR TITLE
merge estimand by estimand

### LINKED ISSUE #304: term
I think we have problems with the current "term" approach for estimates taken from a model

 * First the estimands get odd names (`(Intercept)`) 
 * Second the estimands are  a pain to name  as you need to consult with model output; and the names are and not in the spirit of estimands based on potential outcomes
 * Third it seems to run into issues when combined with other appraoches as shown below
 * Fourth we don't need it. We are already able to do the same thing in a simpler way, see below

Is there anything against the simpler approach below?

```
# The issue:

population  <- declare_population(N = 100, X = rnorm(N))
potentials  <- declare_potential_outcomes(Y ~ (.25 + X) * Z + rnorm(N))
assignment  <- declare_assignment(m = 50)
design_stub <- population +  potentials + assignment + declare_reveal()

estimands_regression <- declare_estimand(
  `(Intercept)` = mean(Y_Z_0),
  `Z` = mean(Y_Z_1 - Y_Z_0),
  term = TRUE,
  label="Regression_Estimands")

estimators_regression <- declare_estimator(
  Y ~ Z,
  estimand = estimands_regression,
  model = lm,
  term = TRUE
)

estimand_2  <- declare_estimand(ATE = mean(Y_Z_1 - Y_Z_0),   label = "ate_simple")
estimator_2 <- declare_estimator(Y~Z, estimand = estimand_2, label = "dim")

design <- design_stub + estimands_regression + estimators_regression  + estimand_2 + estimator_2

# This has four rows instead of 3
simulate_design(design, sims = 1)

# This produces a warning and fails to connect estimates to estimand for dim
diagnose_design(design, sims = 2)

# Note that constituent parts work ok when declared separately
design_A <- design_stub + estimand_2 + estimator_2
diagnose_design(design_A, sims = 2)

design_B <- design_stub + estimands_regression + estimators_regression
diagnose_design(design_B, sims = 2)
```

# A simpler, better approach?

* Getting rid of term from the estimands declaration seems a lot simpler and gives users control over names
* Then in estimates declaration give the estimands of interest. "term" is a bit of a funny term for term. 
* With this usage, the older term "coefficients"  seems more accurate

```

estimands_regression <- declare_estimand(
  Q1 = mean(Y_Z_0),
  Q2 = mean(Y_Z_1 - Y_Z_0),
  label="Regression_Estimands")

estimators_regression <- declare_estimator(
  Y ~ Z,
  estimand = c("Q1", "Q2"),
  model = lm,
  term = TRUE
)

design <- design_stub + estimands_regression + estimators_regression + estimand_2 + estimator_2

# Now fine
simulate_design(design, sims = 1)

# Now fine
diagnose_design(design, sims = 2)

# To boot: nice flexibility to access different estimands 
estimator_3 <- declare_estimator(Y~Z, estimand = c(estimand_2, "Q2"), label = "dim2")
diagnose_design(design + estimator_3,  sims = 20)
```

### PR DESCRIPTION
**Pull request by acoppock**
This should *actually* close #304 .

if the estimates and estimands df all have the same estimand labels, We go through estimand by estimand.

If not, we do the old merge.

### CODE REVIEW

File: R/simulate_design.R [hunk 1]
```diff
@@ -132,46 +132,82 @@ simulate_single_design <- function(design, sims) {
...
-      estimates_df,
-      by = colnames(estimands_df) %icn% estimates_df,
-      all = TRUE,
-      sort = FALSE
-    )
-
-    if (nrow(simulations_df) > max(nrow(estimands_df), nrow(estimates_df))) {
-      warning(
-        "Estimators lack estimand/term labels for matching, a many-to-many merge was performed."
+
```
Comment (nfultz) [inline R/simulate_design.R]: `else if`

File: tests/testthat/test-diagnose-design.R [hunk 2]
```diff
@@ -209,3 +209,35 @@ test_that("with and without term",{
...
+  estimand_2  <- declare_estimand(ATE = 2,   label = "2")
+  estimator_2 <-
+    declare_estimator(Y ~ Z, estimand = estimand_2, label = "dim")
+  
+  design <-
+    population + estimands_regression + estimators_regression + estimand_2 + estimator_2
+  
+  sims_df <- simulate_design(design, sims = 1)
+  
+  expect_equal(nrow(sims_df), 3)
```
Comment (nfultz) [inline tests/testthat/test-diagnose-design.R:240]: Can you add an expectation with more detail - which estimands go with with estimates.

File: R/simulate_design.R (hunk 1 above)
Comment (acoppock) [reply R/simulate_design.R]: thx

File: tests/testthat/test-diagnose-design.R (hunk 2 above)
Comment (acoppock) [reply tests/testthat/test-diagnose-design.R:240]: did a lil dput

**Pull request merged**